Source/monster.cpp

```diff
@@ -4634,23 +4634,48 @@ void ProcessMonsters()
 			Monst->_mAISeed = GetRndSeed();
 		}
 		if (!(monster[mi]._mFlags & MFLAG_NOHEAL) && Monst->_mhitpoints < Monst->_mmaxhp && Monst->_mhitpoints >> 6 > 0) {
-			if (Monst->mLevel <= 1) {
-				Monst->_mhitpoints += Monst->mLevel;
-			} else {
+			if (Monst->mLevel > 1) {
 				Monst->_mhitpoints += Monst->mLevel >> 1;
+			} else {
+				Monst->_mhitpoints += Monst->mLevel;
 			}
 		}
 		mx = Monst->_mx;
 		my = Monst->_my;
 #ifndef SPAWN
+#ifndef HELLFIRE
 		if (dFlags[mx][my] & BFLAG_VISIBLE && Monst->_msquelch == 0 && Monst->MType->mtype == MT_CLEAVER) {
 			PlaySFX(USFX_CLEAVER);
 		}
+#else
+		if (dFlags[mx][my] & BFLAG_VISIBLE && Monst->_msquelch == 0) {
+			if (Monst->MType->mtype == MT_CLEAVER) {
+				PlaySFX(USFX_CLEAVER);
+			}
+			if (Monst->MType->mtype == 137) { //TODO: apply enums
+				if (UseCowFarmer) {
+					PlaySFX(997);
+				} else {
+					int sound = 995;
+					if (!IsUberRoomOpened)
+						sound = 996;
+					PlaySFX(sound);
+				}
+			}
+			if (Monst->MType->mtype == 124)

```

Comment:
```suggestion
			if (Monst->MType->mtype == MT_DEFILER)
```
enums.h line 1685
`	MT_DEFILER  = 0x7C,`